G2P.pm

```diff
@@ -1352,7 +1352,7 @@ sub read_gene_data_from_file {
         }
       }
       else {
-        my %tmp = map {$headers[$_] => $split[$_]} (0..$#split);
+        my %tmp = map { $headers[$_] => (defined $split[$_] ? $split[$_] : '') } (0..$#headers);
```

Comment:
The issue is caused by the comments. Some comments are being splitted into separate rows. In the end, the number of headers and data rows don't match. Using `(0..$#headers)` makes sure the values are assigned correctly but the comment ends up truncated.

We also need to check why the comments keep being splitted. This is the code that needs to be fixed:
```
chomp;
      $_ =~ s/\R//g;
      my @split = grep defined, "$_," =~ /$re/g;
```